packages/liveblocks-react/src/comments/lib/create-cache-manager.ts

```diff
@@ -0,0 +1,49 @@
+import { makeEventSource } from "@liveblocks/core";
+
+import type {
+  Cache,
+  CacheManager,
+  MutationInfo,
+  RequestInfo,
+} from "./revalidation";
+
+export default function createCacheManager<Data>(): CacheManager<Data> & {
```

Comment:
Since they're so tied together, I think we should keep them in a single `cache-manager.ts` file.

Reply:
They are quite relevant to each other. However, the revalidation methods only require the cache manager to be a certain shape. It doesn't really care of the implementation of it. For instance, in this particular cache manager implementation, we have an extra `subscribe` property which isn't relevant at all to the revalidation logic but is only used by our `useThreads` hook.

That being said, I do think there is still a large enough coupling between the two (especially with how strict the shapes are), so it might make sense to keep the two together for now and think of segregating later in the future.

Reply:
I don't think we should focus too much on making it generic tbh, we've (I've 🙈) tried in the past with `AsyncCache` but then use-case A needs batching, use-case B needs optimistic updates' rollbacks while use-case A doesn't, and use-case C (the threads use-case) needs specific optimizations (pagination, granularity, etc). If it's easier for us to build/optimize/maintain 2 or 3 use-case-specific caches I don't think it's too bad.

Reply:
Agreed! My original intention was not to make the code more generic but more from the maintainability perspective. It was just moving the revalidation and mutation logic out of the `createCommentsRoom` method without updating any logic (except for the new network loss condition) so that the code is slightly more maintainable and easier to read. 

I agree it is a very hard balance though.

Reply:
I think the new tests really take advantage of this decoupling as the hooks can be tested independently without the extra overhead of setting up Liveblocks client, creating room context, etc. Here is the test file: [Link](https://github.com/liveblocks/liveblocks/blob/629fa11d181eb07cf01e0f3ad84efa872a6cf911/packages/liveblocks-react/src/__tests__/revalidation.test.tsx)